packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -435,8 +473,8 @@ class RenderGrid extends RenderVirtualViewport<GridParentData> {
       double tileTop = _specification.rowOffsets[placement.row] + gridTopPadding;
       double tileBottom = _specification.rowOffsets[placement.row + placement.rowSpan] + gridTopPadding;
 
-      double childWidth = tileRight - tileLeft - placement.padding.horizontal;
-      double childHeight = tileBottom - tileTop - placement.padding.vertical;
+      double childWidth = math.max(0.0, tileRight - tileLeft - _specification.columnSpacing);
```

Comment:
actually that's not right either. half the column spacing applied to the first and last column, and the whole spacing to the interior columns. but still no spacing if there's only one.